terraform/gcp/k8s/environments/prod.yaml

```diff
@@ -5,23 +5,44 @@ deploymentZone: prod
 replicas: 3
 dsmUrl: https://dsm.datadonationplatform.org/dsm
 dsmIssuer: juniper.terra.bio
-# "portals" adds certificates for each portal - both for the admin subdomains and the custom domain
+# "portals" is used only to generate the admin certificate (without wildcards):
+# e.g. for sandbox.demo.juniper-cmi.org, irb.demo.juniper-cmi.org, etc.
+# any addition to this list will require a short downtime for the admin site while the new
+# certificate is issued.
 portals:
-  - name: demo
-    customDomain: juniperdemostudy.org
-  - name: ourhealth
-    customDomain: ourhealthstudy.org
-  - name: gvasc
-    customDomain: gvascstudy.org
-  - name: atcp
-  - name: hearthive
-    customDomain: thehearthive.org
-  - name: trccproject
-    customDomain: trccproject.org
-  - name: rgp
-  - name: cmi
-  - name: pedihcc
-    customDomain: pedihccproject.org
+  - demo
+  - ourhealth
+  - gvasc
+  - atcp
+  - hearthive
+  - trccproject
+  - rgp
+  - cmi
+  - pedihcc
+
+# we have a maximum of 15 certificates per-ingress. since we have 1 admin certificate,
+# we have a maximum of 14 study certificates in this list.
+# if we need more, we will need to split into multiple ingresses (which costs money).
+# where possible, group multiple portals into a single certificate. this would mean
+# downtime for all URLs in a group when a new URL is added, but that should be rare.
+studyCertificates:
+  - groupName: demo
+    urls:
+      - juniperdemostudy.org
+  - groupName: ourhealth
+    urls:
+      - ourhealthstudy.org
+  - groupName: gvasc
+    urls:
+      - gvascstudy.org
+  - groupName: hearthive
+    urls:
+      - thehearthive.org
+  - groupName: cmi
+    urls:
+      - trccproject.org
+      - pedihccproject.org
```

Comment:
This should work totally fine; waiting to hear back from parker if they are cool with a short downtime for trcc. I told him it was very much not a big deal either way since we have more study slots.

We _could_ buy a second dev URL to test grouping study URLs; it would be nice, but I don't think it's a huge deal. Worst come to worst: we manually fix up the TRCC certificate if grouping causes issues.

Reply:
thinking about it more.... theoretically we could keep the OLD trccproject cert, by doing:
```
  - groupName: trccproject
    urls:
      - trccproject.org
  - groupName: cmi
    urls:
      - trccproject.org
      - pedihccproject.org
```
and then later deleting the original `trccproject` cert; this would have no downtime. i _think_ this would work.

Reply:
set this ^ up plus an extra cert on the demo side for `juniperdemostudy.dev`

if:
- release goes out and no downtime for juniperdemostudy.dev (old cert is not recreated)
- new cert generates as normal
- having 2 certs on demo server does not cause problems

then we should be good to go live with no downtime for any studies, but with a new grouped cert for all CMI studies.